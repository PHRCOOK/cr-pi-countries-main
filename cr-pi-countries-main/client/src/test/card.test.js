
import React from 'react';
import * as router from 'react-router'
import { render, fireEvent,screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Card from '../components/card/card';

const navigate = jest.fn()
beforeEach(() => {
    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate)
  })

  describe('Card', () => {
    it('should render correctly', () => {
      const { container } = render(
        <MemoryRouter>
          <Card id={"ARG"} name={"Argentina"} image={"flag.png"} continent={"America"} />
        </MemoryRouter>
      );
      expect(container).toMatchSnapshot();
    });
  
    it('should render props correctly', () => {
      render(
        <MemoryRouter>
          <Card id={"ARG"} name={"Argentina"} image={"flag.png"} continent={"America"}/>
        </MemoryRouter>
      );
      expect(screen.getByText(/Argentina/i).textContent).toBe("Argentina")
      expect(screen.getByAltText(/bandera/i).getAttribute("src")).toBe("flag.png")
      expect(screen.getByText(/America/i).textContent).toBe("America")
    });
  
  it('should call navigate on click', () => {
    render(
      <MemoryRouter>
        <Card id={"ARG"} name={"Argentina"} navigate={navigate} />
      </MemoryRouter>
    );
    fireEvent.click(screen.getByRole('dialog'));
    expect(navigate).toHaveBeenCalledWith('/detail/ARG');
  });
});