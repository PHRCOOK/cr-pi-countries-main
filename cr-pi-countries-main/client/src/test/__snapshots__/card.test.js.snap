// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div>
  <div
    class="cardContainer"
    role="dialog"
  >
    <img
      alt="bandera del pais"
      class="image"
      src="flag.png"
    />
    <h3
      class="name"
    >
      Argentina
    </h3>
    <h4
      class="continent"
    >
      America
    </h4>
  </div>
</div>
`;
